Apply the Firefox eslint rules, minus the Firefox-specific plugins.

hildjj opened the pull request:
Fix up the myriad changes.  Justification for the large changeset for quotes: they were inconsistent before.

fitzgen commented:
Also, is there a way to enforce using `const` for variables that are not reassigned? If so, we should apply that.

tromey commented:
> Also, is there a way to enforce using const for variables that are not reassigned? If so, we should apply that.

This was recently agreed to in devtools: https://github.com/devtools-html/rfcs/issues/44.
The rule is `prefer-const: "error"`.

hildjj commented:
Note that `npm run lint` now gets called in `npm test`, so travis should pick this up going forward.  See https://travis-ci.org/hildjj/source-map/builds/375031782 for evidence.  I've updated the patch with const/let changes as requested, and added `"no-var": "error"` as well, to make `prefer-const` carry more weight.

Any other eslint rules to add?  May as well put them all into an omnibus patch. :)

fitzgen commented:
LGTM -- do you have anything else @tromey?

Thanks for sticking with this @hildjj!
